addons/website_blog/static/src/js/systray_items/new_content.js

```diff
@@ -6,8 +6,26 @@ patch(NewContentModal.prototype, {
         super.setup();
 
         const newBlogElement = this.state.newContentElements.find(element => element.moduleXmlId === 'base.module_website_blog');
-        newBlogElement.createNewContent = () => this.onAddContent('website_blog.blog_post_action_add', true);
+        newBlogElement.createNewContent = () => this.onAddContent("website_blog.blog_post_action_add", true, this.getCurrentBlogContext());
         newBlogElement.status = MODULE_STATUS.INSTALLED;
         newBlogElement.model = 'blog.post';
     },
+
+    getCurrentBlogContext() {
+        // Using iframe to access mainObject to check if we are on blog page
+        const iframe = document.querySelector("iframe.o_iframe");
+        const isBlogPage = iframe.contentDocument.documentElement.dataset.mainObject?.startsWith("blog");
+
+        if (isBlogPage) {
+            const blogEl = iframe.contentDocument.querySelector(
+                "#wrap.website_blog [data-oe-model='blog.blog']"
+            );
+            const blogId = parseInt(blogEl?.dataset.oeId);
+
+            if (blogId) {
+               return { default_blog_id: blogId };
+            }
+        }
+        return undefined;
+    },
```

Comment:
Sorry, but I'm still not entirely convinced by this approach. What do you think about that:

```suggestion
    getCurrentBlogContext() {
        // Using iframe to access mainObject to check if we are on blog page
        const iframeEl = document.querySelector("iframe.o_iframe");
        const iframeContentDocument = iframeEl.contentDocument || iframeEl.contentWindow.document;
        if (!iframeContentDocument) {
            return;
        }
        const mainObject = iframeContentDocument.documentElement.dataset.mainObject;
        if (mainObject?.startsWith("blog.blog(")) {
            const blogId = parseInt(mainObject.match(/\d+/)?.[0], 10);
            if (!isNaN(blogId)) {
                return { default_blog_id: blogId };
            }
        }
    },
```

Reply:
@SergeBayet,

- The reason why `main-object` cannot be used to parse the ID of the current Blog is because when the user is on a particular blog post, `main-object` holds a record of `blog.post` and not `blog.blog` (see image below). For the same reason, I have only checked for a `blog` match in `main-object`.

![image](https://github.com/user-attachments/assets/8f746923-f1f7-4fff-8632-880874f40628)

- On the blogs page, the `o_wblog_blog_top` element, and on the `blog.post` page, the `o_wblog_post_content` element, both have a reference to the current `blog.blog` record, which is accessed through querySelector, and then the `blogId` is parsed via the dataset.

- For the same reason I think variable names justify the work they are used for.
Thanks.. :smile:

Reply:
Ok but in this case I'd prefer to have a fallback if the page is a `blog.post(` (don't forget the opening parenthesis). We could have a custom/future model `blog.post_admin` or whatever for example.
I could think of an hybrid approach:

```js
getCurrentBlogContext() {
    const iframeEl = document.querySelector("iframe.o_iframe");  // <-- don't forget to add 'El' suffix
    const iframeDoc = iframeEl?.contentDocument || iframeEl?.contentWindow?.document; // retrocompatibility
    if (!iframeDoc) return;

    const mainObject = iframeDoc.documentElement.dataset.mainObject;
    if (mainObject?.startsWith("blog.blog(")) {
        const blogId = parseInt(mainObject.match(/\d+/)?.[0], 10);
        if (!isNaN(blogId)) return { default_blog_id: blogId }; // Check the validity with 'isNaN'
    }

    // Fallback: look in blog post DOM if we're on a blog.post
    if (mainObject?.startsWith("blog.post(")) {
        const el = iframeDoc.querySelector("[data-oe-model='blog.blog']");
        const blogId = parseInt(el?.dataset.oeId);
        if (!isNaN(blogId)) return { default_blog_id: blogId };
    }
}
```

Are you agree with me on this? (I hope so :rofl: )

Reply:
@SergeBayet ,
Even though new models might be introduced, they will eventually start with `blog.model_name`. I believe both approaches work similarly, but writing a separate case for each model may not be optimal. It might be better to generalize the approach, which is what I have already applied in my solution :grin: .